pkg/scheduler/util/non_zero.go

```diff
@@ -38,41 +39,92 @@ const (
 
 // GetNonzeroRequests returns the default cpu and memory resource request if none is found or
 // what is provided on the request.
-func GetNonzeroRequests(requests *v1.ResourceList) (int64, int64) {
-	return GetNonzeroRequestForResource(v1.ResourceCPU, requests),
-		GetNonzeroRequestForResource(v1.ResourceMemory, requests)
+func GetNonzeroRequests(pod *v1.Pod) (int64, int64) {
+	return GetNonzeroRequestForResource(pod, v1.ResourceCPU),
+		GetNonzeroRequestForResource(pod, v1.ResourceMemory)
 }
 
 // GetNonzeroRequestForResource returns the default resource request if none is found or
 // what is provided on the request.
-func GetNonzeroRequestForResource(resource v1.ResourceName, requests *v1.ResourceList) int64 {
-	switch resource {
-	case v1.ResourceCPU:
-		// Override if un-set, but not if explicitly set to zero
-		if _, found := (*requests)[v1.ResourceCPU]; !found {
-			return DefaultMilliCPURequest
+func GetNonzeroRequestForResource(pod *v1.Pod, resource v1.ResourceName) int64 {
+	var podRequest int64
+
+	if len(pod.Spec.Containers) == 0 {
+		return podRequest
+	}
+
+	if resource == v1.ResourceEphemeralStorage && !utilfeature.DefaultFeatureGate.Enabled(features.LocalStorageCapacityIsolation) {
```

Comment:
this is different in https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/framework/v1alpha1/types.go#L203
could you make that consistent?

Reply:
sure